pinot-controller/src/main/java/org/apache/pinot/controller/api/access/AccessControl.java

```diff
@@ -55,10 +40,8 @@ default boolean hasDataAccess(HttpHeaders httpHeaders, String tableName) {
    * @param endpointUrl the request url for which this access control is called
    * @return whether the client has permission
    */
-  default boolean hasAccess(@Nullable String tableName, AccessType accessType, HttpHeaders httpHeaders,
-      @Nullable String endpointUrl) {
-    return true;
-  }
+  boolean hasAccess(@Nullable String tableName, AccessType accessType, HttpHeaders httpHeaders,
```

Comment:
Why does this no longer default `true`? This is backwards incompatible for users of this interface

Reply:
Good catch. Adding it back in #14382